src/components/MessagesList/PinnedMessage/PinnedMessage.vue

```diff
@@ -0,0 +1,195 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<script setup lang="ts">
+
+import { t } from '@nextcloud/l10n'
+import { computed, onMounted } from 'vue'
+import { useRoute } from 'vue-router'
+import { useStore } from 'vuex'
+import NcActionButton from '@nextcloud/vue/components/NcActionButton'
+import NcListItem from '@nextcloud/vue/components/NcListItem'
+import IconClose from 'vue-material-design-icons/Close.vue'
+import IconPinOff from 'vue-material-design-icons/PinOffOutline.vue'
+import AvatarWrapper from '../../AvatarWrapper/AvatarWrapper.vue'
+import { AVATAR } from '../../..//constants.ts'
+import { useGetToken } from '../../../composables/useGetToken.ts'
+import { EventBus } from '../../../services/EventBus.ts'
+import { useActorStore } from '../../../stores/actor.ts'
+import { useSharedItemsStore } from '../../../stores/sharedItems.ts'
+import { parseToSimpleMessage } from '../../../utils/textParse.ts'
+
+const store = useStore()
+const actorStore = useActorStore()
+const route = useRoute()
+const token = useGetToken()
+const sharedItemsStore = useSharedItemsStore()
+const conversation = computed(() => store.getters.conversation(token.value))
+const richSubline = computed(() => {
+	if (!pinnedMessage.value) {
+		return ''
+	}
+	return parseToSimpleMessage(
+		pinnedMessage.value.message,
+		pinnedMessage.value.messageParameters,
+	)
+})
+
+// Array of all pinned messages
+const pinnedMessages = computed(() => {
+	if (!sharedItemsStore.sharedItems(token.value).pinned) {
+		return []
+	}
+	return Object.values(sharedItemsStore.sharedItems(token.value).pinned) || []
+})
+
+// The pinned message to be displayed (the latest one that is not hidden)
+const pinnedMessage = computed(() => {
+	if (!pinnedMessages.value.length) {
+		return null
+	}
+	const item = pinnedMessages.value.find((item) => +item.id === conversation.value.lastPinnedId)
+
+	if (!item) {
+		// Id not found (e.g: deleted, expired), return most recent pinned message
+		const fallbackId = sharedItemsStore.findMostRecentPinnedMessageId(token.value)
+		return pinnedMessages.value.find((item) => +item.id === fallbackId)
+	}
+	return item.id !== conversation.value.hiddenPinnedId ? item : null
+})
+
+// Display name for the pinned message
+const displayNameDetails = computed(() => {
+	const isPinnerSameAsAuthor = pinnedMessage.value?.metaData?.pinnedActorId === pinnedMessage.value?.actorId
```

Comment:
I'd add this to get rid of all questionmarks. Otherwise it's not a pinned message, right?
```suggestion
	if (!pinnedMessage.value?.metaData) {
		return ''
	}
	const isPinnerSameAsAuthor = pinnedMessage.value.metaData.pinnedActorId === pinnedMessage.value.actorId
```